### ISSUE TITLE
Support TypeScript 5.0

### ISSUE THREAD

**Issue by sosukesuzuki** [prettier/prettier#14240]
We should support new syntax will be landed in [TS 5.0](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0-beta/).

**parsers status:**
- typescript-eslint
  - https://github.com/typescript-eslint/typescript-eslint/issues/6380
- Babel
  - https://github.com/babel/babel/issues/15374
  - https://github.com/prettier/prettier/issues/14390

## [Decorators](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0-beta/#decorators)

```ts
class Person {
    name: string;
    constructor(name: string) {
        this.name = name;
    }

    @loggedMethod
    greet() {
        console.log(`Hello, my name is ${this.name}.`);
    }
}

const p = new Person("Ray");
p.greet();
```

## [`const` Type Parameters](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0-beta/#const-type-parameters)

```ts
type HasNames = { names: readonly string[] };
function getNamesExactly<const T extends HasNames>(arg: T): T["names"] {
    return arg.names;
}
```

## [`export type *`](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0-beta/#support-for-export-type)

```ts
export type * as vehicles from "./spaceship";
```

**Comment by fisker** [comment]
Build script update #13819

**Comment by nandorojo** [comment]
Sorry to ping this issue, but to confirm, is there an experimental version published anywhere that has TypeScript 5.0 support yet? Couldn't find it in the releases, so figured I would ask just in case. 

Judging by the checklist at the top: Babel support has been merged, and TypeScript ESlint support is released, albeit somewhat experimental. The TS ESLint version has been working for me, though.

Thanks!

**Comment by sosukesuzuki** [comment]
> Sorry to ping this issue, but to confirm, is there an experimental version published anywhere that has TypeScript 5.0 support yet? 

No, we haven't published the release that supports TypeScript 5.0. We will let you know in this issue when we release it, so please wait.

**Comment by sssyed** [comment]
hey folks sorry to bump this again, since https://github.com/prettier/prettier/pull/13819 has been merged and [TS 5.0](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/) is out will we see a release soon?

**Comment by fisker** [comment]
https://github.com/prettier/prettier/releases/tag/2.8.5
